js/product-details.js

```diff
@@ -0,0 +1,161 @@
+    activateFirstButton()
+  }
+};
+xmlhttp.open("GET", "../jsons/" + productCategory + ".json", true);
+xmlhttp.send();
+
+const activateFirstButton = function () {
+  document.getElementsByClassName("details-description")[0].classList.add("details-button-active--row");
+  document.getElementsByClassName("product-details-value--row")[0].classList.toggle("product-details-value--row--none");
+}
+
+const buildDetailButton = function (drink) {
+  var productDetails = document.getElementsByClassName("product-details")[0];
+  let detailButton = document.createElement("div");
+  detailButton.innerHTML = key;
+  detailButton.classList.add("details-" + key);
+  detailButton.classList.add("details-button");
+  productDetails.appendChild(detailButton);
+};
+
+const bildDetailvaluecontainers = function (drink) {
+  var containerColumn = document.getElementsByClassName("product-details")[0];
+  var containerRow = document.getElementsByClassName(
+    "product-details-value"
+  )[0];
+  if (typeof drink[key] == "object" && !Array.isArray(drink[key])) {
+    if (drink.hasOwnProperty(key)) {
+      ifValueObject(drink[key], containerColumn, containerRow)
+    }
+  } else if (Array.isArray(drink[key])) {
+    ifValueArray(drink[key], containerColumn, containerRow)
+  } else {
+    ifValueString(drink[key], containerColumn, containerRow);
+  }
+};
+
+const ifValueObject = function (value, containerColumn, containerRow) {
+  let drinkKey = value;
+  let valuecontainerColumn = document.createElement("div");
+  let valuecontainerRow = document.createElement("div");
+  let key = "";
+  for (key in drinkKey) {
+    if (drinkKey.hasOwnProperty(key)) {
+      showKey(key, valuecontainerColumn)
+      showKey(key, valuecontainerRow)
+      if (Array.isArray(drinkKey[key])) {
+        forEachArrayElement(drinkKey[key], valuecontainerColumn, valuecontainerRow);
+      } else {
+        showArrayElement(drinkKey[key], valuecontainerColumn)
+        showArrayElement(drinkKey[key], valuecontainerRow)
+      }
+    }
+  }
+  containerColumn.appendChild(addPlacementClassesToElement(valuecontainerColumn, "column"));
+  containerRow.appendChild(addPlacementClassesToElement(valuecontainerRow, "row"));
+}
+
+const ifValueArray = function (value, containerColumn, containerRow) {
+  let valuecontainerColumn = document.createElement("div");
+  let valuecontainerRow = document.createElement("div");
+  forEachArrayElement(value, valuecontainerColumn, valuecontainerRow);
+  containerColumn.appendChild(addPlacementClassesToElement(valuecontainerColumn, "column"));
+  containerRow.appendChild(addPlacementClassesToElement(valuecontainerRow, "row"));
+}
+
+const forEachArrayElement = function (value, valuecontainerColumn, valuecontainerRow) {
+  value.forEach(function (element) {
+    showArrayElement(element, valuecontainerColumn)
+    showArrayElement(element, valuecontainerRow)
+  })
+}
+
+const ifValueString = function (value, containerColumn, containerRow) {
```

Comment:
what is this function for?
just call showStringValue in place which you call ifValueString